sangyeob/app.js

```diff
@@ -51,11 +50,179 @@ app.post('/users', async (req, res) => {
+        `SELECT
+            users.id as userId,
+            users.profile_image as userProfileImage,
+            posts.id as postingId,
+            posts.content_image as postingImageUrl,
+            posts.content as postingContent
+        FROM users
+        INNER JOIN posts ON posts.user_id = users.id
+        `,
+        (err, posts) => {
+            return res.status(200).json(posts);
+        }
+    );
+});
+
+app.get('/users/:userId/posts', async (req, res) => {
+    const { userId } = req.params;
+
+    const obj = {};
+    obj.userId = userId;
+    try {
+        let userProfileImage = await database.query(
+            `SELECT
+                profile_image
+            FROM users 
+            WHERE id = ${userId};
+            `
+        );
+        const { profile_image } = JSON.parse(
+            JSON.stringify(userProfileImage[0])
+        );
+
+        obj.userProfileImage = profile_image;
+
+        let postsbyUser = await database.query(
+            `SELECT
+                id as postingId,
+                content_image as postingImageUrl,
+                content as postingContent
+            FROM posts 
+            WHERE user_id = ${userId};
+            `
+        );
+
+        obj.postings = postsbyUser;
+        return res.status(200).json({ data: obj });
+    } catch (err) {
+        console.log(err);
+        return res.status(409).json({ error: 'invalid input' });
+    }
+});
+
+app.put('/posts', async (req, res) => {
+    const { postingId, postingTitle, postingContent, postingImage } = req.body;
+    try {
+        await database.query(
+            `UPDATE posts
+                SET
+                    title = ?,
+                    content = ?,
+                    content_image = ?
+                    WHERE id = ?;
+            `,
+            [postingTitle, postingContent, postingImage, postingId]
+        );
+        await database.query(
+            `SELECT
+                users.id as userId,
+                users.name as userName,
+                posts.id as postingId,
+                posts.title as postingTitle,
+                posts.content as postingContent
+            FROM users
+            INNER JOIN posts 
+            WHERE posts.id = ${postingId} AND posts.user_id=users.id;
+            `,
+            (err, rows) => {
+                return res.status(200).json({ data: rows });
+            }
+        );
+    } catch (err) {
+        return res.status(409).json({ error: err.sqlMessage });
+    }
+});
+
+app.delete('/posts/:postId', async (req, res) => {
+    const { postId } = req.params;
+    await database.query(
+        `DELETE FROM likes
+        WHERE post_id = ${postId}
+        `
+    );
+    await database.query(
+        `DELETE FROM posts
+		WHERE id = ${postId}
+		`
+    );
+    return res.status(200).json({ message: 'successfully deleted' });
+});
+
+app.post('/likes/:userId/:postId', async (req, res) => {
+    const { userId, postId } = req.params;
```

Comment:
```suggestion
app.post('/likes', async (req, res) => {
    const { userId, postId } = req.body;
```

likes API는 userId, postId를 request body로 받을 수 있게 구현해주세요!